services/common/src/components/documents/spatial/AddSpatialDocumentsModal.tsx

```diff
@@ -95,18 +99,35 @@ const AddSpatialDocumentsModal: FC<AddSpatialDocumentsModalProps> = ({
             Please confirm that the composed map below correctly represents your selected spatial
             data files.
           </Typography.Paragraph>
-          <ViewSpatialDetail spatialDocuments={existingDocuments} />
+          {filesHaveGeomarkIds() && <ViewSpatialDetail spatialDocuments={existingDocuments} />}
         </>
       ),
     },
   ];
 
+  const addBundleIdToFiles = ({
+    docman_bundle_guid,
+    geomark_id,
+  }: {
+    docman_bundle_guid: string;
+    geomark_id: string;
+  }) => {
+    const newFiles = currentValues[fieldName].map((f) => {
+      return {
+        ...f,
+        docman_bundle_guid,
+        geomark_id,
+      };
+    });
+    dispatch(change(modalFormName, fieldName, [...initialDocuments, ...newFiles]));
+  };
+
   const handleSubmit = async (values) => {
     const isFinalStep = currentStep === stepContent.length - 1;
     const newFiles = values[fieldName];
-    const allFiles = [...newFiles, ...initialDocuments];
     if (isFinalStep) {
-      dispatch(change(formName, fieldName, allFiles));
+      dispatch(change(formName, fieldName, [...initialDocuments, ...newFiles]));
+      dispatch(change(formName, "spatial_documents", [...initialDocuments, ...newFiles]));
```

Comment:
Was this second dispatch added by mistake?

Reply:
Indeed!